lib/assertions.js

```diff
@@ -1107,17 +1107,23 @@ module.exports = function (expect) {
         this.errorMode = 'nested';
         var that = this;
         return subject.then(function (value) {
-            return that.shift(value, 0);
-        }, function (err) {
-            if (typeof nextAssertion === 'string') {
-                that.argsOutput = function (output) {
-                    output.error(nextAssertion);
-                    var rest = that.args.slice(1);
-                    if (rest.length > 0) {
-                        output.sp().appendItems(rest, ', ');
-                    }
-                };
+            if (expect.findTypeOf(nextAssertion).is('expect.it')) {
+                // Force a failing expect.it error message to be property nested instead of replacing the default error message:
```

Comment:
I don't think it would, it seems like there's an `expect.it` quirk in play here. Ideally shifting to an `expect.it` and an assertion string should behave the same. If you check out the branch and remove the special casing of the `expect.it` here, the assertion inside the `should fail with the right error message when the next assertion is an expect.it that fails` test fails like this:

```
expected { foo: 'bar' } to have property 'foo' with a value of 'quux'

-bar
+quux
```

But I want it to fail like this:

```
expected Promise when fulfilled expect.it('to have property', 'foo', 'quux')
  expected { foo: 'bar' } to have property 'foo' with a value of 'quux'

  -bar
  +quux
```